### ISSUE TITLE
Crash on Startup

### ISSUE THREAD

**Issue by JackCorsair** [endless-sky/endless-sky#5887]
I open the game, get a black screen, and then a notification that the game has stopped working and needs to close.

**To Reproduce**
1. Open game
2. Game crashes
2. Cry

**Expected behavior**
Ideally, the game opens and I get to play more of this wonderful love letter to EV.

**System (please complete the following information):**
 - OS: Windows 10 Home, version 10.0.19042, build 19042
 - Game Source: Steam
 - Version: Build ID 6558675

**Comment by tehhowch** [comment]
There isn't enough here to go on yet, unfortunately. Would you share information about installed plugins (if any), the contents of `errors.txt` (in your `%appdata%\endless-sky` directory), your hardware (CPU + GPU, RAM amount), and possibly a screen recording of the launch process?

If at all possible, opening a console window and launching the game's .exe from that, with the `-p` flag, would be helpful too.

**Comment by JackCorsair** [comment]
The issue seems to have fixed itself? I gave up, played some fallout 4 for a while, then tried again and it started up just fine. In the interests of being thorough, however:

CPU: Intel Core i7-6700k
GPU: NVidia GeForce GTX1070
RAM: 16GB

Contents of errors.txt:
Warning: government " Drak " is referred to, but never defined.
Warning: government " Hai " is referred to, but never defined.
Warning: government " Independent " is referred to, but never defined.
Warning: government " Ka'het " is referred to, but never defined.
Warning: government " Kor Mereti " is referred to, but never defined.
Warning: government " Scar's Legion " is referred to, but never defined.
Warning: system "Membulem" is referred to, but never defined.

**Comment by Water261** [comment]
> The issue seems to have fixed itself? I gave up, played some fallout 4 for a while, then tried again and it started up just fine. In the interests of being thorough, however:
> 
> CPU: Intel Core i7-6700k
> GPU: NVidia GeForce GTX1070
> RAM: 16GB
> 
> Contents of errors.txt:
> Warning: government " Drak " is referred to, but never defined.
> Warning: government " Hai " is referred to, but never defined.
> Warning: government " Independent " is referred to, but never defined.
> Warning: government " Ka'het " is referred to, but never defined.
> Warning: government " Kor Mereti " is referred to, but never defined.
> Warning: government " Scar's Legion " is referred to, but never defined.
> Warning: system "Membulem" is referred to, but never defined.
> 
> Edit: It's worth noting that I only posted here after several reinstallations, computer restarts, and verification of local files through Steam.

Try downloading the same build from GitHub and see what changes. It may be a problem with Steam.

**Comment by JackCorsair** [comment]
I did try that before I posted. It gave me the same problem. All the more baffling that it just... sorted itself out after a while.

**Comment by principis** [comment]
I have the same issue with the steam version of the game on Fedora 34. The  save was previously used in 0.9.12 but trying to open in 0.9.13.

```
Warning: government " Drak " is referred to, but never defined.
Warning: government " Hai " is referred to, but never defined.
Warning: government " Independent " is referred to, but never defined.
Warning: government " Ka'het " is referred to, but never defined.
Warning: government " Kor Mereti " is referred to, but never defined.
Warning: government " Scar's Legion " is referred to, but never defined.
Warning: government " Drak " is referred to, but never defined.terminate called after throwing an instance of 'std::bad_cast'
  what():  std::bad_cast
```

It works however with the version from the Fedora repositories:
```
Warning: government " Drak " is referred to, but never defined.
Warning: government " Hai " is referred to, but never defined.
Warning: government " Independent " is referred to, but never defined.
Warning: government " Ka'het " is referred to, but never defined.
Warning: government " Kor Mereti " is referred to, but never defined.
Warning: government " Scar's Legion " is referred to, but never defined.
Warning: government " Drak " is referred to, but never defined.
Warning: government " Hai " is referred to, but never defined.
Warning: government " Independent " is referred to, but never defined.
Warning: government " Ka'het " is referred to, but never defined.
Warning: government " Kor Mereti " is referred to, but never defined.
Warning: government " Scar's Legion " is referred to, but never defined.
```

**Comment by tehhowch** [comment]
Please post the save.

From that trace, the value passed to Files::LogError is of interest (frames 7 and 8)

**Comment by principis** [comment]
Here is the save: [save.txt](https://github.com/endless-sky/endless-sky/files/6623122/save.txt)
I'm unable to retrieve the values `No symbol table info available.` or at least I don't know how...

**Comment by tehhowch** [comment]
I can reproduce with the provided save in the steam-shipped v0.9.13, thanks.

Similarly, a self-compiled v0.9.13 checkout works ok. Will debug

**Comment by tehhowch** [comment]
What's interesting to me is that this only happens if we're using a statically linked libstdc++, which is done solely for the Steam Linux build. That is, if I build the game inside a containerized Steam runtime using the default build instructions, launching the binary with your save results in the following trace:

```
Thread 1 "es-steam-debug" received signal SIGABRT, Aborted.
0x00007f851f6c0035 in raise () from /lib/x86_64-linux-gnu/libc.so.6
(gdb) bt
#0  0x00007f851f6c0035 in raise () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007f851f6c379b in abort () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x000000000075a975 in __gnu_cxx::__verbose_terminate_handler() ()
#3  0x000000000072c746 in __cxxabiv1::__terminate(void (*)()) ()
#4  0x000000000072c773 in std::terminate() ()
#5  0x000000000072c6e2 in __cxa_throw ()
#6  0x00000000007572d2 in std::__throw_bad_cast() ()
#7  0x00000000007332ad in std::basic_ostream<char, std::char_traits<char> >& std::endl<char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&) ()
#8  0x000000000044bbc6 in Files::LogError (
    message="Endless Sky requires OpenGL version 3.0 or higher.\nYour OpenGL version is 2.1 Mesa 8.0.4, GLSL version 1.20.\nPlease update your graphics drivers.") at source/Files.cpp:553
#9  0x00000000004a78fb in GameWindow::ExitWithError (
    message="Endless Sky requires OpenGL version 3.0 or higher.\nYour OpenGL version is 2.1 Mesa 8.0.4, GLSL version 1.20.\nPlease update your graphics drivers.", doPopUp=true) at source/GameWindow.cpp:409
#10 0x00000000004a713a in GameWindow::Init () at source/GameWindow.cpp:192
#11 0x000000000040cf7c in main (argc=1, argv=0x7ffceceb8218) at source/main.cpp:126
```
Note that the bad_cast is being thrown from `cerr << message << endl;`, and specifically the `<< endl;` flush.

If we modify the build instructions to use the dynamically linked libstdc++, then there is no `bad_cast` thrown, and launching the game from inside the steam runtime Docker container (correctly) quits with the message about too low of an OpenGL version. If I then exfiltrate this binary from the container and replace my steam library binary with this self-compiled one, it works just fine via Steam.

Removing the static library inclusion won't fix the OP's issue (which occurred on Windows), but will provide some stability to others.

**Comment by principis** [comment]
@tehhowch thank you! :) Sorry for the thread hijacking, I thought it was the same issue :)

**Comment by Wedge009** [comment]
> Contents of errors.txt:
Warning: government " Drak " is referred to, but never defined.
Warning: government " Hai " is referred to, but never defined.
Warning: government " Independent " is referred to, but never defined.
Warning: government " Ka'het " is referred to, but never defined.
Warning: government " Kor Mereti " is referred to, but never defined.
Warning: government " Scar's Legion " is referred to, but never defined.
Warning: system "Membulem" is referred to, but never defined.

When I started playing self-compiled 0.9.14, I saw similar warning messages in stdout/stderr and I assumed they were due to bugs in older versions. Removing those weird government entries got rid of those warnings and I fixed Membulem manually - it was renamed in #5328. Likewise I had warnings from the removal of the Queri Quarg Ringworld in #4921 so I had to remove my visitation of the now non-existent 'planet'.

I never had any crashes, though. I started briefly on Ubuntu's [packaged 0.9.8](https://packages.ubuntu.com/focal/endless-sky), then switched to Steam 0.9.12 release, then moved to self-compiled 0.9.14 before I started noticing these warnings.
